### PR TITLE
Linkchecker update

### PR DESCRIPTION
**Pull request by SimonGrn**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | linkchecker refactorisation : new reporting system, new place (in `tools`), better error catching
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| How to test?  | npm run linkchecker

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15980)

### CODE REVIEW

File: tests/puppeteer/tools/urls.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+module.exports = [
+  /* {
```
Comment (PierreRambaud) [inline tests/puppeteer/tools/urls.js]: Remove useless comment

Comment (SimonGrn) [reply tests/puppeteer/tools/urls.js]: Done, forget i did that to accelerate my tests
Thanks

**Review by PierreRambaud — APPROVED** [review]

File: tests/puppeteer/README.md [hunk 2]
```diff
@@ -68,7 +68,14 @@ This script will detect not found and erroneous pages, by crawling your back off
 
 
 ### Launch script
-If you want to run the links checker test you can run the script **`campaigns/linkchecker.js`**
+If you want to run the links checker test you can run the script **`tools/linkchecker.js`**.
+It uses a `urls.js` file describing all the URLs it can crawl.
+
+You **must** disable the Security Token before running this script ! Add this line in your `.htacess` file:
```
Comment (Progi1984) [inline tests/puppeteer/README.md]:
```suggestion
You **must** disable the Security Token before running this script ! Add this line in your `.htaccess` file:
```

Comment (PierreRambaud) [reply tests/puppeteer/README.md]: nice catch

File: tests/puppeteer/tools/linkchecker.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+const filename = `report_${dateString}_${hours}-${minutes}-${seconds}`;
+
+
+const URL_FO = process.env.URL_FO || 'http://localhost/prestashop/';
+const URL_BO = process.env.URL_BO || `${URL_FO}admin-dev/`;
+const LOG_PASSED = process.env.LOG_PASSED || false;
+const HEADLESS = process.env.HEADLESS || true;
+
+const LOGININFOS = {
+  user: {
```
Comment (Progi1984) [inline tests/puppeteer/tools/linkchecker.js:22]: may be client rather than user

Comment (PierreRambaud) [reply tests/puppeteer/tools/linkchecker.js:22]: It's also a user :)

Comment (Progi1984) [reply tests/puppeteer/tools/linkchecker.js:22]: An admin is also a user :)

Comment (PierreRambaud) [reply tests/puppeteer/tools/linkchecker.js:22]: yep but an admin is a super user :trollface:

Comment (Progi1984) [reply tests/puppeteer/tools/linkchecker.js:22]: So it's a user :trollface:

Comment (PierreRambaud) [reply tests/puppeteer/tools/linkchecker.js:22]: But is it also a client? :thinking: 
Should be call them, `client_user_admin_super_user` :fearful:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Typo unresolved

File: tests/puppeteer/tools/linkchecker.js (hunk 3 above)
Comment (SimonGrn) [reply tests/puppeteer/tools/linkchecker.js:22]: It's for making a difference between the admin in the BO and the user in the FO.

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**